src/peripherals/using-interior-mutability.md

```diff
@@ -0,0 +1,96 @@
+# Interior Mutability
+
+What's the matter of using `static mut` variables anyway?
+
+In Rust, whenever we have a variable `foo`, only one of the following may happen:
+
+* the variable `foo` is moved to another place. Subsequent reference of `foo` would be a compile failure as the value has been moved. 
+* some number of `&foo` borrows, which allow read-only access.
+* a single (exclusive) `&mut foo` mutable borrow, which allow read/write. 
+
+One consequence of a `static` variable is that it cannot be moved, as it lives _statically_ in memory, meaning it is in a fixed place.
+
+A `static` variable can be borrowed as `&foo` without much issue as it is just reading.
+
+The problem occurs when we try to borrow is as mutable. Consider the following:
+
+```rust,ignore
+static mut flag: bool = false;
+
+
+fn first() {
+    if !flag {
+        flag = true;
+        println!("first");
+    }
+}
+
+fn second() {
+    if !flag {
+        flag = true;
+        println!("second");
+    }
+}
+```
+
+The question is, which one would run? It's not obviousy just looking at a code. Perhaps it depends on the order or execution, which would be guaranteed _if_ there's only one active execution at a time.
+
+You may think that because you're in an embedded programming land and you only have single core, you don't have to think about race condition, but even with a single core CPU without making any thread, you may have interrupts and those interrupt requests may access the shared mutable variable `flag`. The issue here is that the compiler cannot prohibit you from taking multiple mutable reference to the shared mutable variable at compile time, especially since it does not know about the order of executions of threads or interrupts, for example.
+
+Also, the use of `static mut` variables _might_ be deprecated in the future. There's a strong indication of this and having the usage `deny` by default since Rust 2024 is one of them.
+
+## So what is interior mutability?
+
+Interior mutability is a way to "trick" the borrow checker to mutate a borrowed reference `&foo` (notice the lack of `&mut`). This is basically us saying to the compiler "I'm going to mutate the shared reference anyway, please don't stop me.".
```

Comment:
```suggestion
Interior mutability is a way around the borrow checker to mutate a borrowed reference `&foo` (notice the lack of `&mut`). This is basically us saying to the compiler "I'm going to mutate the shared reference anyway, please don't stop me.".
```
I think "trick" sounds a bit too alluring :) Since we do not want people to use unsafe more that really strictly necessary, I would make the wording slightly more neutral.